### PR TITLE
add solution1

### PR DESCRIPTION
**Pull request by evgenkolner**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -51,6 +50,16 @@
             <artifactId>mysql-connector-j</artifactId>
             <scope>runtime</scope>
         </dependency>
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-core</artifactId>
+            <version>${liquibase.version}</version>
```
Comment (Elena-Bruyako) [inline pom.xml]: redundant. spring-boot-starter-parent already includes the version of liquibase

File: src/main/java/bookstore/repository/BookRepository.java [hunk 2]
```diff
@@ -1,13 +1,9 @@
...
 import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
 
-public interface BookRepository {
-    Book save(Book book);
-
-    List<Book> findAll();
-
-    Optional<Book> getBookByID(Long id);
+public interface BookRepository extends JpaRepository<Book, Long> {
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/repository/BookRepository.java:9]: annotation Repository is missing

File: src/main/java/bookstore/service/BookServiceImpl.java [hunk 3]
```diff
@@ -31,9 +31,20 @@ public List<BookDto> findAll() {
...
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto bookRequestDto) {
+        Book book = bookMapper.toBook(bookRequestDto);
+        book.setId(id);
+        return bookMapper.toDto(bookRepository.save(book));
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/service/BookServiceImpl.java:51]: To make logic correct, add method to BookMapper
void updateBook(CreateBookRequestDto updatedRequestDto, @MappingTarget Book bookToUpdate); 

Then here, find book by Id in DB, use this updateBook method 
 bookMapper.updateBook(updateBookRequestDto, bookToUpdate);
 
 and then convert toDto

File: src/main/java/bookstore/controller/BookController.java [hunk 4]
```diff
@@ -18,19 +22,34 @@
...
         return bookService.getBookById(id);
     }
 
+    @ResponseStatus(HttpStatus.CREATED)
     @PostMapping
     public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
 
+    @ResponseStatus(HttpStatus.NO_CONTENT)
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/controller/BookController.java:43]: Not correct status. 
OK will be better

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

**Review by Andriy-dz — APPROVED** [review]

**Pull request merged**